marshal/tinyint/unmarshal_utils.go

```diff
@@ -22,11 +23,19 @@ func DecInt8(p []byte, v *int8) error {
 }
 
 func DecInt8R(p []byte, v **int8) error {
-	if p != nil {
-		*v = new(int8)
-		return DecInt8(p, *v)
+	switch len(p) {
+	case 0:
+		if p == nil {
+			*v = nil
+		} else {
+			*v = new(int8)
+		}
+	case 1:
+		val := int8(p[0])
+		*v = &val
```

Comment:
Question, it will panic for `(**int8)(nil)`, is it ok ?

Reply:
Yeap, same as any try to dereference nil reference

Reply:
Fixed